scadalts-ui/src/views/Reports/ReportsPage.vue

```diff
@@ -0,0 +1,941 @@
+				v => 
+				this.reportForm.schedule && !!v || this.$t('reports.minutesRequired'),
+				v => 
+				this.reportForm.schedule && !isNaN(v*1) || v *1 > 0 || this.$t('reports.delayMustBePositive') ,
+			],
+			emailRules: [
+				v => 
+				(!v || this.validateEmail(v)) || this.$t('reports.emailMustBeValid'),
+			],
+			cronRules: [
+				v => 
+				this.reportForm.schedule && !!v || this.$t('reports.cronPatternIsRequired'),
+				v =>
+				/(@(annually|yearly|monthly|weekly|daily|hourly|reboot))|(@every (\d+(ns|us|µs|ms|s|m|h))+)|((((\d+,)+\d+|(\d+(\/|-)\d+)|\d+|\*) ?){5,7})/.test(v) ||
+				$t('reports.cronPatternMustBeValid')
+			],
+			
+			periodTypes:[ 
+				{value: 2, text:this.$t("common.timeperiod.minutes")},
+				{value: 3, text:this.$t("common.timeperiod.hours")},
+				{value: 4, text:this.$t("common.timeperiod.days")},
+				{value: 5, text:this.$t("common.timeperiod.weeks")},
+				{value: 6, text:this.$t("common.timeperiod.months")},
+				{value: 7, text:this.$t("common.timeperiod.years")},
+			],
+			periodTypesPlusCron:[ {value: 0, text: `cron`},
+				{value: 2, text:this.$t("common.timeperiod.minutes")},
+				{value: 3, text:this.$t("common.timeperiod.hours")},
+				{value: 4, text:this.$t("common.timeperiod.days")},
+				{value: 5, text:this.$t("common.timeperiod.weeks")},
+				{value: 6, text:this.$t("common.timeperiod.months")},
+				{value: 7, text:this.$t("common.timeperiod.years")},
+			],
+			userList: [],
+			selectedUser: null,
+			email: '',
+			
+			alarmOptions: [
+				{ label:this.$t("reports.events.none"), value: 1 },
+				{ label:this.$t("reports.events.alarms"), value: 2 }, 
+				{ label:this.$t("reports.events.all"), value: 3 },
+				
+			],
+			dateRangeTypeOptions: [
+				{ label: this.$t("reports.specificDates"), value:2 },
+				{ label: this.$t("reports.relative"), value:1 }
+			],
+			snackbarMessage: '',
+			snackbar: false,
+			dialog: false,
+			search: '',
+			searchDatapointsKeywords: '',
+			searchDatapoints: [],
+			reportList: [],
+			datapointToSave: null,
+			selectedDatapointId: null,
+			datapoints: [],
+			options: {},
+			datapointOptions: {},
+			selectedId: -1,
+			selectedReport: null,
+			reportForm: {
+				points: [],
+				relativeRangePreviousQuantity: 0,
+				dateRangeType: 1,
+				includeUserComments: false,
+				schedule: false,
+ 
+            	name: '', 
+            	id:-1,
+				points:[], 
+				includeEvents:1, 
+				relativeDateType:0,
+				previousPeriodCount:0,
+            	previousPeriodType:0,
+				pastPeriodCount:0,
+				pastPeriodType:0,
+				fromNone:false,
+				fromYear:0,
+            	fromMonth:0,
+				fromDay:0,
+				fromHour:0,
+				fromMinute:0,
+				toNone:false,
+				toYear:0,
+				toMonth:0,
+            	toDay:0,
+				toHour:0,
+				toMinute:0,
+				schedule:false,
+				schedulePeriod:0,
+				runDelayMinutes:0,
+            	scheduleCron:'',
+				email:false,
+				includeData:false,
+				zipData:false,
+            	recipients:[],//List<RecipientListEntryBean> 
+			},
```

Comment:
Maybe it will be nice to move such a big object to a separate file? You can create a JavaScript class definition with all that default values initialized and the you can just initialize that object here? 
```
import ReportForm form './models/ReportForm.js'
...
reportForm: new ReportForm(),
```
In this way you can reduce the size of this file and also organize your code. But this is also only a suggestion

Also that could be useful because in line 770 you are using the `createNewReport()` method that is doing the same as you did here - it is initializing that object with the same parameters. So if you have two places where you initialize the similar object you should consider to moving that thing to separate file.